pennylane/ops/functions/dot.py

```diff
@@ -89,18 +89,21 @@ def dot(
     # When casting a Hamiltonian to a Sum, we also cast its inner Tensors to Prods
     ops = [qml.prod(*op.obs) if isinstance(op, Tensor) else op for op in ops]
 
-    if coeffs[0] != 1 and qml.math.allequal(coeffs[0], coeffs):
+    if not qml.math.allclose(coeffs[0], 1) and qml.math.allequal(coeffs[0], coeffs):
```

Comment:
do you know what caused this inaccuracy?

Reply:
So I'm actually not getting test failures when I revert this. So 🤞 it turns out I didn't actually need to do this?